src/Microsoft.Azure.WebJobs/FunctionNameAttribute.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.WebJobs
+{
+    /// <summary>
+    /// This is for providing a custom name to for an SDK function.
```

Comment:
Also, we'll want to verify that things work end to end when this is applied in the traditional webjobs Dashboard. E.g. if you override the display name, that name is shown, and everything still continues to work - Dashboard lookups etc. still function.